### PR TITLE
[terraform-client] print only to file that is not in a git repository

### PR DESCRIPTION
**Pull request by maorfr**
part of https://issues.redhat.com/browse/APPSRE-4208

this will prevent terraform printed configuration files from being leaked in an accidental commit

### CODE REVIEW

File: reconcile/utils/git.py [hunk 1]
```diff
@@ -25,3 +25,10 @@ def checkout(commit, wd):
     proc.communicate()
     if proc.returncode != 0:
         raise GitError('git checkout failed: {}'.format(commit))
+
+
+def is_file_in_git_repo(file_path):
+    real_path = os.path.realpath(file_path)
+    dir_path = os.path.dirname(real_path)
+    git_path = os.path.join(dir_path, '.git')
+    return os.path.exists(git_path)
```
Comment (geoberle) [inline reconcile/utils/git.py:34]: this only catches paths that are in the root of qontract-reconcile but not ones in subdirectories.

Comment (maorfr) [reply reconcile/utils/git.py]: fixed in https://github.com/app-sre/qontract-reconcile/pull/2066/commits/a5e879e254ea434254c7b71f667a840b2aa87cd5

File: reconcile/utils/git.py [hunk 2]
```diff
@@ -30,5 +30,8 @@ def checkout(commit, wd):
 def is_file_in_git_repo(file_path):
     real_path = os.path.realpath(file_path)
     dir_path = os.path.dirname(real_path)
-    git_path = os.path.join(dir_path, '.git')
-    return os.path.exists(git_path)
+    with open(os.devnull, 'w') as dev_null:
+        proc = Popen(['git', 'git rev-parse', '--is-inside-work-tree'],
```
Comment (geoberle) [inline reconcile/utils/git.py]: the first parameter to the git command should be only `rev-parse` not `git rev-parse`

Comment (maorfr) [reply reconcile/utils/git.py]: can't a guy make typos in peace? :wink: 
fixed in https://github.com/app-sre/qontract-reconcile/pull/2066/commits/825c961a1ce9e3afa76f8807032793424012cd12

Comment (geoberle) [reply reconcile/utils/git.py]: No 😛

**Review by geoberle — APPROVED** [review]
LGTM

File: reconcile/utils/git.py [hunk 3]
```diff
@@ -25,3 +25,14 @@ def checkout(commit, wd):
...
+
+
+def is_file_in_git_repo(file_path):
+    real_path = os.path.realpath(file_path)
+    dir_path = os.path.dirname(real_path)
+    # pylint: disable=consider-using-with
+    DEVNULL = open(os.devnull, 'w')
+    proc = Popen(['git', 'rev-parse', '--is-inside-work-tree'],
+                 cwd=dir_path, stdout=DEVNULL, stderr=DEVNULL)
+    proc.communicate()
```
Comment (Piojo) [inline reconcile/utils/git.py:37]: It's easier to just use `subprocess.PIPE` here instead of that `devnull` redirection.

**Review by Piojo — COMMENTED** [review]
I try not to be a pain around unit tests for `{ŧerraform,terrascript}_client` but this one is the result of a security problem. Please add a couple of tests for `dump`.

File: reconcile/utils/git.py (hunk 3 above)
Comment (maorfr) [reply reconcile/utils/git.py:37]: as this is a security problem, i'm going to merge it before i start working on tests i will never be able to complete in a timely manner.

Comment (maorfr) [reply reconcile/utils/git.py:37]: also note that the only change i made to `ts.dump` is tested, so :shrug:

Comment (maorfr) [reply reconcile/utils/git.py:37]: https://github.com/app-sre/qontract-reconcile/pull/2085/commits/ccd0397437ed1de451dc6d9a9c9fa7ab0e0314b4

**Pull request merged**